specs/agents/tracing-instrumentation-db.md

```diff
@@ -1,34 +1,198 @@
-## Database spans
+## Database and Datastore spans
 
-We capture spans for various types of database/data-stores operations, such as SQL queries, Elasticsearch queries, Redis commands, etc. We follow some of the same conventions defined by OpenTracing for capturing database-specific span context, including:
+We capture spans for various types of database/data-stores operations, such as SQL queries, Elasticsearch queries, Redis commands, etc. 
+Database and datastore spans **must not have child spans** (within the same transaction), e.g. they must be exit spans for the service.
 
- - `db.instance`: database instance name, e.g. "customers". For DynamoDB, this is the region.
- - `db.statement`: statement/query, e.g. "SELECT * FROM foo"
- - `db.user`: username used for database access, e.g. "readonly_user"
- - `db.type`: database type/category, which should be "sql" for SQL databases, and the lower-cased database name otherwise.
+The following fields are relevant for database and datastore spans. Where possible, agents should provide information for as many as possible of these fields. The semantics of and concrete values for these fields may vary between different technologies. See sections below for details on specific technologies.
 
-The full database statement should be stored in `db.statement`, which may be useful for debugging performance issues. We store up to 10000 Unicode characters per database statement.
+| Field | Description | Mandatory |
```

Comment:
I don't know if practical/feasible, but it would help *me* to if something indicated the impact of having or not having a field. Specifically: what parts of the user experience (is the UX in APM UI the only user experience here?) uses a particular field.

Reply:
will followup on this idea with a separate PR